Felienne/hedy
# [UI IMPROVEMENT] Makes color of modal alert dependent of message

**Description**
In this PR we improve the modal alert implementation to support both green and red messages. This way we can return a more visual feedback to the user if the alert is successful or not. Currently all modal alert messages are green, but this might be confusing when an error is returned. We add an additional boolean parameter to the alert to define if a message is an error or not. We also implement a dedicated modal for the repair program. This to better differentiate between the "intrusive" modals and the alert one.

**Fix for**
This PR fixes #1763.

**How to test**
We have to test three things to make sure it works:

- Perform a successful action (for example, save or delete a program) -> should be green
- Perform an error action (try to rename a class to an empty string) -> should be red
- Call the repair modal (create a faulty program and call the repair logo) -> should be the "intrusive" modal
